aws/resource_aws_emr_instance_group.go

```diff
@@ -85,197 +126,196 @@ func resourceAwsEMRInstanceGroup() *schema.Resource {
+		for i, rule := range ig.AutoScalingPolicy.Rules {
+			for j, dimension := range rule.Trigger.CloudWatchAlarmDefinition.Dimensions {
+				if *dimension.Key == "JobFlowId" {
+					tmpDimensions := append(ig.AutoScalingPolicy.Rules[i].Trigger.CloudWatchAlarmDefinition.Dimensions[:j], ig.AutoScalingPolicy.Rules[i].Trigger.CloudWatchAlarmDefinition.Dimensions[j+1:]...)
+					ig.AutoScalingPolicy.Rules[i].Trigger.CloudWatchAlarmDefinition.Dimensions = tmpDimensions
+				}
+			}
+
+			if len(ig.AutoScalingPolicy.Rules[i].Trigger.CloudWatchAlarmDefinition.Dimensions) == 0 {
+				ig.AutoScalingPolicy.Rules[i].Trigger.CloudWatchAlarmDefinition.Dimensions = nil
+			}
+		}
+
+		autoscalingPolicyConstraintsBytes, err := json.Marshal(ig.AutoScalingPolicy.Constraints)
+		if err != nil {
+			return fmt.Errorf("error parsing EMR Cluster Instance Groups AutoScalingPolicy Constraints: %s", err)
+		}
+
+		autoscalingPolicyRulesBytes, err := normalizeEmptyRules(ig.AutoScalingPolicy.Rules)
+		if err != nil {
+			return fmt.Errorf("error parsing EMR Cluster Instance Groups AutoScalingPolicy Rules: %s", err)
+		}
+
+		autoscalingPolicyString = fmt.Sprintf("{\"Constraints\":%s,\"Rules\":%s}", string(autoscalingPolicyConstraintsBytes), string(autoscalingPolicyRulesBytes))
+	}
+	d.Set("autoscaling_policy", autoscalingPolicyString)
+
+	if ig.BidPrice != nil {
+		d.Set("bid_price", ig.BidPrice)
+	}
+
+	d.Set("instance_count", ig.RequestedInstanceCount)
+	d.Set("instance_role", ig.InstanceGroupType)
+	d.Set("instance_type", ig.InstanceType)
+	d.Set("name", ig.Name)
+	d.Set("running_instance_count", ig.RunningInstanceCount)
+
+	if ig.Status != nil && ig.Status.State != nil {
+		d.Set("status", ig.Status.State)
 	}
 
 	return nil
 }
 
-func fetchAllEMRInstanceGroups(conn *emr.EMR, clusterId string) ([]*emr.InstanceGroup, error) {
-	req := &emr.ListInstanceGroupsInput{
-		ClusterId: aws.String(clusterId),
-	}
+func resourceAwsEMRInstanceGroupUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).emrconn
 
-	var groups []*emr.InstanceGroup
-	marker := aws.String("intitial")
-	for marker != nil {
-		log.Printf("[DEBUG] EMR Cluster Instance Marker: %s", *marker)
-		respGrps, errGrps := conn.ListInstanceGroups(req)
-		if errGrps != nil {
-			return nil, fmt.Errorf("Error reading EMR cluster (%s): %s", clusterId, errGrps)
+	log.Printf("[DEBUG] Modify EMR task group")
+	if d.HasChange("instance_count") {
+		instanceCount := d.Get("instance_count").(int)
+
+		params := &emr.ModifyInstanceGroupsInput{
+			InstanceGroups: []*emr.InstanceGroupModifyConfig{
+				{
+					InstanceGroupId: aws.String(d.Id()),
+					InstanceCount:   aws.Int64(int64(instanceCount)),
+				},
+			},
 		}
-		if respGrps == nil {
-			return nil, fmt.Errorf("Error reading EMR Instance Group for cluster (%s)", clusterId)
+
+		_, err := conn.ModifyInstanceGroups(params)
+		if err != nil {
+			return err
```

Comment:
We should return error context here for operators and code maintainers:

```suggestion
			return fmt.Errorf("error modifying EMR Instance Group (%s): %s", d.Id(), err)
```